plugins/3dtext.lua

```diff
@@ -82,16 +82,14 @@ else
 			local object = nut.markup.parse("<font=nut3D2DFont>"..text:gsub("\\n", "\n"))
 			-- We want to draw a shadow on the text object.
 			object.onDrawText = function(text, font, x, y, color, alignX, alignY, alpha)
-				surface.SetTextPos(x+1, y+1)
-				surface.SetTextColor(0, 0, 0, alpha)
-				surface.SetFont(font)
-				surface.DrawText(text)
-
-				surface.SetTextPos(x, y)
-				surface.SetTextColor(color.r, color.g, color.b, alpha)
-				surface.SetFont(font)
-				surface.DrawText(text)
-				--draw.SimpleTextOutlined(text, font, x, y, ColorAlpha(color, alpha), alignX, alignY, 2, color_black)
+				draw.TextShadow({
```

Comment:
Can we create a table outside the function and just update the keys accordingly inside? That way, we won't be creating a lot of tables every frame. This would probably also help with garbage collection.